src/sentry/constants.py

```diff
@@ -76,6 +76,9 @@ def get_all_languages():
 MAX_CULPRIT_LENGTH = 200
 MAX_EMAIL_FIELD_LENGTH = 75
 
+ENVIRONMENT_NAME_PATTERN = r'^[a-z0-9][a-z0-9_\-\. :]*$'
```

Comment:
Not that I’m disagreeing, but this pattern is now significantly more restrictive. This is going to start rejecting a lot more environment names than we are rejecting today. 

So without evaluating how many this will affect, I know it’ll be a lot, I think we should just keep it to just not allowing newlines or slashes.

Reply:
Indeed it would. After reading the paper doc you sent me I thought that the pattern outlined there was the desired state.

I've scaled back the validation to only disallow `/` and newline characters.